sqldb/sqlc/queries/graph.sql

```diff
@@ -132,3 +132,46 @@ SELECT sn.node_id, n.pub_key
 FROM source_nodes sn
     JOIN nodes n ON sn.node_id = n.id
 WHERE n.version = $1;
+
+/* ─────────────────────────────────────────────
+   channels table queries
+   ─────────────────────────────────────────────
+*/
+
+-- name: CreateChannel :one
+INSERT INTO channels (
```

Comment:
We could consider a more complex query that's less roundtrips and less fragile as well than executing get then insert if the channel not exist. Something like this should do the trick:

```
-- name: CreateChannel :one
WITH try_insert AS (
    INSERT INTO channels (
        version, scid, node_id_1, node_id_2,
        outpoint, capacity, bitcoin_key_1, bitcoin_key_2,
        node_1_signature, node_2_signature, bitcoin_1_signature,
        bitcoin_2_signature
    )
    SELECT
        $1, $2, $3, $4, $5, $6, $7, $8,
        $9, $10, $11, $12
    WHERE NOT EXISTS (
        SELECT 1 FROM channels WHERE scid = $2
    )
    RETURNING id, false AS existed
)
SELECT id, existed FROM try_insert
UNION ALL
SELECT id, true AS existed FROM channels WHERE scid = $2
LIMIT 1;
```